docs/streams/developer-guide/dsl-api.html

```diff
@@ -3097,152 +3098,994 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
+            &quot;omg&quot;, &quot;oh my god&quot;,
+            &quot;btw&quot;, &quot;by the way&quot;
+    );
+    private static final String INPUT_MESSAGES_TOPIC = &quot;input-messages-topic&quot;;
+    private static final String OUTPUT_MESSAGES_TOPIC = &quot;output-messages-topic&quot;;
+
+    public static void replaceWithProcessValues(final StreamsBuilder builder) {
+        KStream&lt;String, String&gt; messageStream = builder.stream(INPUT_MESSAGES_TOPIC);
+        messageStream.processValues(SlangReplacementProcessor::new).to(OUTPUT_MESSAGES_TOPIC);
+    }
+
+    private static class SlangReplacementProcessor extends ContextualFixedKeyProcessor&lt;String, String, String&gt; {
+        @Override
+        public void process(final FixedKeyRecord&lt;String, String&gt; record) {
+            if (record.value() == null) {
+                return; // Skip null values
+            }
+
+            // Replace slang words in the message
+            final String[] words = record.value().split(&quot;\\s+&quot;);
+            for (final String word : words) {
+                String replacedWord = SLANG_DICTIONARY.getOrDefault(word, word);
+                context().forward(record.withValue(replacedWord));
+            }
+        }
+    }
+}</code></pre>
+                <h4>Keynotes</h4>
+                <ul>
+                    <li><strong>Type Safety and Flexibility:</strong> The process and processValues APIs utilize
+                        <code>ProcessorContext</code> and <code>Record</code> or <code>FixedKeyRecord</code> objects for better type
+                        safety and flexibility of custom processing logic.
+                    </li>
+                    <li><strong>Clear State and Logic Management:</strong> Implementations for <code>Processor</code> or
+                        <code>FixedKeyProcessor</code> should manage state and logic clearly. Use <code>context().forward()</code>
+                        for emitting records downstream.
+                    </li>
+                    <li><strong>Unified API:</strong> Consolidates multiple methods into a single, versatile API.</li>
+                    <li><strong>Future-Proof:</strong> Ensures compatibility with the latest Kafka Streams releases.</li>
+                </ul>
+            </div>
+            <div class="section" id="transformers-removal-and-migration-to-processors">
+                <a class="headerlink" href="#transformers-removal-and-migration-to-processors" title="Permalink to this headline">
+                    <h3>
+                        <a class="toc-backref" href="#id37">Transformers removal and migration to processors</a>
+                    </h3>
+                </a>
+                <p>As of Kafka 4.0, several deprecated methods in the Kafka Streams API, such as <code>transform</code>,
+                    <code>flatTransform</code>,<br><code>transformValues</code>, <code>flatTransformValues</code>, and
```

Comment:
```suggestion
                    <code>flatTransform</code>, <code>transformValues</code>, <code>flatTransformValues</code>, and
```

Reply:
There is more `<br>` tags below -- seems they should all be replace why a space?

Reply:
Replaced.